packages/instrumentation-aws-lambda/test/lambda-test/sync.js

```diff
@@ -15,26 +15,40 @@
  */
 const api = require('@opentelemetry/api');
 
-exports.handler = function (event, context, callback) {
-  callback(null, 'ok');
+// Promise-based handlers (all Node.js versions)
+exports.handler = async function (event, context) {
+  return 'ok';
 };
 
-exports.error = function (event, context, callback) {
+exports.error = async function (event, context) {
   throw new Error('handler error');
 };
 
-exports.callbackerror = function (event, context, callback) {
-  callback(new Error('handler error'));
+exports.callbackerror = async function (event, context) {
+  throw new Error('handler error');
+};
+
+exports.stringerror = async function (event, context) {
+  throw 'handler error';
 };
 
-exports.stringerror = function (event, context, callback) {
+exports.callbackstringerror = async function (event, context) {
```

Comment:
nit: I suppose the "callback" name in this test fixture is a bit out of place now.